### PR TITLE
Enhanced the DXO data and meta default to empty dict.

### PR DESCRIPTION
**Pull request by yhwen**

### CODE REVIEW

File: nvflare/apis/dxo.py [hunk 1]
```diff
@@ -47,10 +47,15 @@ def __init__(self, data_kind: str, data: dict, meta: dict = None):
...
-            data_kind:
+            data_kind: data_kind name
             data: clear-text data
             meta: None or dict
         """
+        if data is None:
+            data = {}
+        if meta is None:
+            meta = {}
+
```
Comment (yanchengnv) [inline nvflare/apis/dxo.py:58]: Since we are doing this, we should make sure data and meta are of dict type.

**Review by yanchengnv — CHANGES_REQUESTED** [review]

File: nvflare/apis/dxo.py (hunk 1 above)
Comment (yhwen) [reply nvflare/apis/dxo.py:58]: We have this:

def __init__(self, data_kind: str, data: dict, meta: dict = None):

**Review by yanchengnv — CHANGES_REQUESTED** [review]

File: nvflare/apis/dxo.py (hunk 1 above)
Comment (yhwen) [reply nvflare/apis/dxo.py:58]: Added data and meta dict type check.

Comment (yhwen) [reply nvflare/apis/dxo.py:58]: Actually we already have the dict type check in:

err = self.validate()

Comment (yanchengnv) [reply nvflare/apis/dxo.py:58]: You are right!

**Review by yanchengnv — APPROVED** [review]

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**